### ISSUE TITLE
TypeScript reformat

### ISSUE THREAD

**Issue by PatrykKuniczak** [bishnudev01/editable-creatable-multiselect#9]
Hi

Your package looks good, but i want to use it in TypeScript project, can u create a update, which allows to use it with TS.

A lot of people coding in TS, because it's very useful :)

**Comment by bishnudev01** [comment]
Will do, thank you!

**Comment by PatrykKuniczak** [comment]
And you must create this React 18 friendly, this React relased 2 days ago but on refactor it's required :)

If u looking for contributor i can help you, but i typing in functions react, IMO it's better way :)

I have a idea for 1 next feature, deleting items not from selected but from e.p. Data Base this is must have, in my current project i want component like this, but with editting, adding and deleting

My idea is add a button on list item like trash this button will deleting item hard, this will be a next callback.

**Comment by PatrykKuniczak** [comment]
We can refactor whole code, and add new features :)

**Comment by PatrykKuniczak** [comment]
2 more suggestions, on Example code, don't give console.log and demo didn't work

**Comment by bishnudev01** [comment]
> And you must create this React 18 friendly, this React relased 2 days ago but on refactor it's required :)
> 
> If u looking for contributor i can help you, but i typing in functions react, IMO it's better way :)
> 
> I have a idea for 1 next feature, deleting items not from selected but from e.p. Data Base this is must have, in my current project i want component like this, but with editting, adding and deleting
> 
> My idea is add a button on list item like trash this button will deleting item hard, this will be a next callback.

yes functional component sounds great too. We have to reformat the whole code. Let's define what changes we want, then we can choose the feature we want to work on.

**Comment by bishnudev01** [comment]
> 2 more suggestions, on Example code, don't give console.log and demo didn't work

yup, it's been crashing on codesandbox ever since I released v 2.1.0
works fine when I install it on my computer or (gcloud for that matter) any idea? 



Regarding console.log, I added it so that user knows what's happening with the lists.

**Comment by PatrykKuniczak** [comment]
> > And you must create this React 18 friendly, this React relased 2 days ago but on refactor it's required :)
> > If u looking for contributor i can help you, but i typing in functions react, IMO it's better way :)
> > I have a idea for 1 next feature, deleting items not from selected but from e.p. Data Base this is must have, in my current project i want component like this, but with editting, adding and deleting
> > My idea is add a button on list item like trash this button will deleting item hard, this will be a next callback.
> 
> yes functional component sounds great too. We have to reformat the whole code. Let's define what changes we want, then we can choose the feature we want to work on.

I start to defining features here: https://github.com/PatrykKuniczak/Notification_Manager/issues/9
Let's read, and invite my on DC: Destrox#2843

**Comment by PatrykKuniczak** [comment]
> > 2 more suggestions, on Example code, don't give console.log and demo didn't work
> 
> yup, it's been crashing on codesandbox ever since I released v 2.1.0 works fine when I install it on my computer or (gcloud for that matter) any idea?
> 
> Regarding console.log, I added it so that user knows what's happening with the lists.

Yeah, but user can do it if want, i don't see a lot of console logs in docs overall
